File: src/main/scala/Day01.scala
class Day01 extends Day {
  override def run(): Unit = {

    val instructions = getResourceAsString("/day01.txt").map(c => if(c.equals('(')) 1 else -1)

    def findIndex: Int = {
      var floor = 0
      var index = 0
      instructions.foreach(i => {
        index = index + 1
        floor = floor + i
        if(floor < 0) {
          return index
        }
      })

      -1
    }

    printAnswer(1, "One", instructions.sum)
    printAnswer(1, "Two", findIndex) //1797
  }
}

File: src/main/java/com/nibado/projects/advent/Day01.java
package com.nibado.projects.advent;

import static com.nibado.projects.advent.Util.printAnswer;
import static com.nibado.projects.advent.Util.readResourceAsString;

public class Day01 implements Runnable {

    public static void main(String... argv) {
        new Day01().run();
    }

    @Override
    public void run() {
        String input = readResourceAsString("/day01.txt");

        int floor = 0;
        for(int i = 0; i < input.length();i++) {
            if(input.charAt(i) == '(') {
                floor++;
            }
            else if(input.charAt(i) == ')') {
                floor--;
            }
        }

        printAnswer(1, "One", floor);

        int i;
        floor = 0;
        for(i = 0; i < input.length();i++) {
            if(input.charAt(i) == '(') {
                floor++;
            }
            else if(input.charAt(i) == ')') {
                floor--;
            }
            if(floor < 0) {
                break;
            }
        }

        printAnswer(1, "Two", i + 1);
    }
}
